activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -26,13 +26,12 @@ module AttributeAssignment
     #   cat.name # => 'Gorby'
     #   cat.status # => 'sleeping'
     def assign_attributes(new_attributes)
-      if !new_attributes.respond_to?(:stringify_keys)
+      unless new_attributes.respond_to?(:stringify_keys)
         raise ArgumentError, "When assigning attributes, you must pass a hash as an argument, #{new_attributes.class} passed."
       end
       return if new_attributes.empty?
 
-      attributes = new_attributes.stringify_keys
-      _assign_attributes(sanitize_for_mass_assignment(attributes))
+      _assign_attributes(sanitize_for_mass_assignment(new_attributes.dup))
```

Comment:
Do we need to `dup` here? It seems we are not modifying it. We could `dup` on activerecord/lib/active_record/attribute_assignment.rb's `_assign_attributes`

Reply:
I don't think we do. That makes sense.